### DISCUSSION TITLE
Action button on Tables - how to disable a button row by row

### DISCUSSION THREAD

**Discussion by russell-hutson-jp** [ToolJet/ToolJet#7402] [Q&A] [answered]
I have a table
![image](https://github.com/ToolJet/ToolJet/assets/102313014/211922bd-0f44-4cc5-9f56-fe642aef360a)

Each row has a button.
I want to disable the button based on a query. so I have a Fx condition on the Disable
![image](https://github.com/ToolJet/ToolJet/assets/102313014/f741066b-056a-4640-a358-b450d82d6d72)

In my query I have
![image](https://github.com/ToolJet/ToolJet/assets/102313014/9668d22e-2d29-4645-a413-e44d086a7357)

which I want to take the '**envelopeid**' from the table and search my database for a match.
If there is a match, the button should be disabled.

The problem is
1. It doesn't fire the query - I have to have an event on the Table to run the query.
2. when it does fire the query, it enables or disables ALL buttons, not only the ones already found in the DB

**Comment by ShazanRizvi** [comment, marked as answer]
Hey @russell-hutson-jp 
You can use `{{rowData}}` to control the enable and disable of the action buttons on the table and this will be according to the data in that particular row, You can get more info about the same in our [documentation.](https://docs.tooljet.com/docs/how-to/access-cellvalue-rowdata/#using-rowdata-to-change-column-text-color)
